tests/test_mapping_json.py

```diff
@@ -0,0 +1,112 @@
+    return Compose(
+        [
+            LoadImage(image_only=True),
+            SaveImage(output_dir=temp_dir, output_ext=".nii.gz", savepath_in_metadict=savepath_in_metadict),
+            MappingJson(mapping_json_path=mapping_json_path),
+        ]
+    )
+
+
+def process_image(args):
+    temp_dir, mapping_json_path, i = args
+    time.sleep(random.uniform(0, 0.1))
+    input_file = create_input_file(temp_dir, f"test_image_{i}")
+    transform = create_transform(temp_dir, mapping_json_path)
+    transform(input_file)
+    time.sleep(random.uniform(0, 0.1))
+
+
+@unittest.skipUnless(has_nib, "nibabel required")
+class TestMappingJson(unittest.TestCase):
+    def setUp(self):
+        self.temp_dir = tempfile.mkdtemp()
+        self.mapping_json_path = os.path.join(self.temp_dir, "mapping.json")
+
+    def tearDown(self):
+        shutil.rmtree(self.temp_dir)
+
+    @parameterized.expand([(True,), (False,)])
+    def test_mapping_json(self, savepath_in_metadict):
+        name = "test_image"
+        input_file = create_input_file(self.temp_dir, name)
+        output_file = os.path.join(self.temp_dir, name, name + "_trans.nii.gz")
+
+        transform = create_transform(self.temp_dir, self.mapping_json_path, savepath_in_metadict)
+
+        if savepath_in_metadict:
+            transform(input_file)
+            self.assertTrue(os.path.exists(self.mapping_json_path))
+            with open(self.mapping_json_path) as f:
+                mapping_data = json.load(f)
+            self.assertEqual(len(mapping_data), 1)
+            self.assertEqual(mapping_data[0]["input"], input_file)
+            self.assertEqual(mapping_data[0]["output"], output_file)
+        else:
+            with self.assertRaises(RuntimeError) as cm:
+                transform(input_file)
+            cause_exception = cm.exception.__cause__
+            self.assertIsInstance(cause_exception, KeyError)
+            self.assertIn(
+                "Missing 'saved_to' key in metadata. Check SaveImage savepath_in_metadict.", str(cause_exception)
+            )
+
+    def test_multiprocess_mapping_json(self):
+        num_processes, num_images = 3, 50
+
+        with multiprocessing.Pool(processes=num_processes) as pool:
+            args = [(self.temp_dir, self.mapping_json_path, i) for i in range(num_images)]
+            pool.map(process_image, args)
+
+        with open(self.mapping_json_path) as f:
+            mapping_data = json.load(f)
+
+        self.assertEqual(len(mapping_data), num_images, f"Expected {num_images} entries, but got {len(mapping_data)}")
+        unique_entries = {tuple(sorted(entry.items())) for entry in mapping_data}
+        self.assertEqual(len(mapping_data), len(unique_entries), "Duplicate entries exist")
+        for entry in mapping_data:
+            self.assertIn("input", entry, "Entry missing 'input' key")
+            self.assertIn("output", entry, "Entry missing 'output' key")

```

Comment:
Another possible way of checking this is to generate a json file with only one process, then do things again with multiple processes and compare the contents.